### PR TITLE
fixes rms.RMSD: selections are now applied to atomgroup itself

### LINKED ISSUE #1684: RMSD select is incorrectly applied to the universe and not the atomgroup
### Expected behaviour
In [analysis.rms.RMSD](https://www.mdanalysis.org/mdanalysis/documentation_pages/analysis/rms.html#MDAnalysis.analysis.rms.RMSD) the *select* string should always be applied to the *atomgroup*, i.e., `atomgroup.select_atoms(select)`.

### Actual behaviour
The selection string is applied to the whole `Universe`, i.e., the equivalent of `atomgroup.universe.atoms.select_atoms(select)`.

In the case where the atomgroup is just `universe.atoms`, the results are the same, otherwise they are different (and wrong).

### Code to reproduce the behaviour

``` python
import numpy as np
import MDAnalysis as mda
from MDAnalysis.analysis import rms
from MDAnalysis.tests.datafiles import PSF, DCD

u = mda.Universe(PSF, DCD)
R = rms.RMSD(u.atoms, select="resid 1-30").run()
R2 = rms.RMSD(u.atoms.select_atoms("name CA"), select="resid 1-30").run()
print(np.allclose(R2.rmsd[:, 2], R.rmsd[:, 2]))
# True
```
This should give `False` because the RMSD of all atoms in residues 1-30 is different from the RMSD of only the C-alpha atoms in residues 1-30.

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)

0.17.0-dev

### PR DESCRIPTION
**Pull request by orbeckst**
Fixes #1684

Changes made in this Pull Request:
 - Previously, using an atomgroup with select was broken (the select string was applied to atomgroup.universe). Now: selections are now applied to atomgroup itself

Note: for group_selections, the selection strings ARE still applied      to the whole atomgroup.universe for maximum flexibility

Tests are still missing.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
I am surprised that the existing tests pass... I still have to raise a proper issue and write tests.

**Comment by kain88-de** [comment]
@orbeckst maybe the tests are wrong?

**Comment by richardjgowers** [comment]
not looked at the tests, but if they operate on `u.atoms` then this change won't have affected them?

File: package/MDAnalysis/analysis/rms.py [hunk 1]
```diff
@@ -350,7 +350,10 @@ def __init__(self, atomgroup, reference=None, select='all',
             described under :ref:`ordered-selections-label`).
 
         groupselections : list (optional)
-            A list of selections as described for `select`. Each selection
+            A list of selections as described for `select`, with the differemce that
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/rms.py]: spelling

**Pull request merged**